### PR TITLE
[IC-365]  Add UpsertMessageStatus implementation

### PR DESCRIPTION
**Pull request by gquadrati**
#### List of Changes

- Add UpsertMessageStatus implementation
- Add tests
- add integration tests
- remove __integrations__ folder from dist folder and eslint checks

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: UpsertMessageStatus/handler.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+export function UpsertMessageStatusHandler(
+  messageStatusModel: MessageStatusModel
+): IUpsertMessageStatusHandler {
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  return async (context, fiscalCode, messageId, change) =>
+    pipe(
+      messageStatusModel.findLastVersionByModelId([messageId]),
+      TE.mapLeft(err => ResponseErrorQuery("findLastVersionByModelId", err)),
+      TE.chainW(
+        // If no message-status was found, build a new one
```
Comment (AleDore) [inline UpsertMessageStatus/handler.ts]: comment to remove :)

File: UpsertMessageStatus/handler.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        TE.fromOption(() =>
+          ResponseErrorNotFound(
+            `Cannot found message status`,
+            `Cannot found message status for message ${messageId}`
+          )
+        )
+      ),
+      TE.chainW(
+        // If no message-status was found, build a new one
```
Comment (AleDore) [inline UpsertMessageStatus/handler.ts]: ditto :)

Comment (gquadrati) [reply UpsertMessageStatus/handler.ts]: Done in [061a5f1](https://github.com/pagopa/io-functions-app-messages/pull/10/commits/061a5f11d3b96e7529d3a73cd0c88cfc1571d445)

File: UpsertMessageStatus/handler.ts (hunk 1 above)
Comment (gquadrati) [reply UpsertMessageStatus/handler.ts]: Done in [061a5f1](https://github.com/pagopa/io-functions-app-messages/pull/10/commits/061a5f11d3b96e7529d3a73cd0c88cfc1571d445)

File: __integrations__/__tests__/upsertMessageStatus.test.ts [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+  QueueStorageConnection,
+  MESSAGE_CONTAINER_NAME
+} from "../env";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import {
+  MessageStatus,
+  MessageStatusModel
+} from "@pagopa/io-functions-commons/dist/src/models/message_status";
+import { MessageStatusChange } from "@pagopa/io-functions-commons/dist/generated/definitions/MessageStatusChange";
+import { randomInt } from "crypto";
```
Comment (AleDore) [inline __integrations__/__tests__/upsertMessageStatus.test.ts]: This import causes fail on integration tests step.

Comment (gquadrati) [reply __integrations__/__tests__/upsertMessageStatus.test.ts]: Thanks. Done in [b1f6e3c](https://github.com/pagopa/io-functions-app-messages/pull/10/commits/b1f6e3c9ff9ce32ae5d3cdcc7417238b04b46171)

**Review by AleDore — APPROVED** [review]

**Pull request merged**